src/Compressors/sampling.jl

```diff
@@ -0,0 +1,226 @@
+    n_rows = size(A, 2)
+    n_cols = compression_dim
+    initial_size = n_rows
+    return n_rows, n_cols, initial_size
+end
+
+function complete_compressor(sub_sampling::Sampling, A::AbstractMatrix)
+    n_rows, n_cols, _ = get_dims(sub_sampling.compression_dim, sub_sampling.cardinality, A)
+    # Pull out the variables from ingredients
+    compression_dim = sub_sampling.compression_dim
+    sub_sampling.distribution.cardinality = sub_sampling.cardinality
+    # Compute the weight for each index
+    dist_recipe = complete_distribution(sub_sampling.distribution, A)
+    idx = Vector{Int64}(undef, compression_dim)
+    idx_v = view(idx,:)
+    # Randomly generate samples from index set based on weights
+    sample_distribution!(idx, dist_recipe)
+    return SamplingRecipe{typeof(sub_sampling.cardinality)}(sub_sampling.cardinality, 
+                                                            compression_dim, 
+                                                            n_rows, 
+                                                            n_cols, 
+                                                            dist_recipe, 
+                                                            idx, 
+                                                            idx_v)
+end
+
+function update_compressor!(S::SamplingRecipe, x::AbstractVector, A::AbstractMatrix, b::AbstractVector)
+    update_distribution!(S.distribution_recipe, x, A, b)
+    # Randomly generate samples from index set based on weights
+    sample_distribution!(S.idx_v, S.distribution_recipe)
+end
+    
+# Matrix-matrix multiplication
+# Begin with the left version
+function mul!(
+    C::AbstractArray, 
+    S::SamplingRecipe{Left}, 
+    A::AbstractArray, 
+    alpha::Number, 
+    beta::Number
+)
+    # Check the compatability of the sizes of the things being multiplied
+    left_mul_dimcheck(C, S, A)
+    for i in 1:length(S.idx_v)
+        c_row = view(C, i, :)
+        a_row = view(A, S.idx[i], :)
+
+        c_row .= beta .* c_row .+ alpha .* a_row
+    end
+
+    return nothing
+end
+
+function mul!(
+    C::AbstractArray, 
+    S::SamplingRecipe{Right}, 
+    A::AbstractArray, 
+    alpha::Number, 
+    beta::Number
+)
+    # Check the compatability of the sizes of the things being multiplied
+    left_mul_dimcheck(C, S, A)
+    C .*= beta
+
+    for i in 1:length(S.idx_v)
+        c_row = view(C, S.idx[i], :)
+        a_row = view(A, i, :)
+
+        c_row .+= alpha .* a_row
+    end
+
+    return nothing
+end
+
+# right version
+function mul!(
+    C::AbstractArray, 
+    A::AbstractArray, 
+    S::SamplingRecipe{Right}, 
+    alpha::Number, 
+    beta::Number
+)
+    # Check the compatability of the sizes of the things being multiplied
+    right_mul_dimcheck(C, A, S)
+    for i in 1:length(S.idx_v)
+        c_col = view(C, :, i)
+        a_col = view(A, :, S.idx[i])
+        
+        c_col .= beta .* c_col .+ alpha .* a_col
+    end
```

Comment:
Try to see if you can just pass this to mul

Reply:
This comment has not been addressed. If you tried this and it did not work, please reply with a comment and mark the comments as resolved.